### PR TITLE
[Prefix] Prefix Upload nit fixes and tests

### PR DESCRIPTION
**Pull request by shishirx34**
Addressing Damon's comments after the previous PR merge, missed passing the flag correctly in one place `GeneratePackageAsync`. Added case-insensitivity test cases. Plus minor nit fixes.

### CODE REVIEW

File: src/NuGetGallery/Services/PackageUploadService.cs [hunk 1]
```diff
@@ -24,15 +24,15 @@ public class PackageUploadService: IPackageUploadService
 
         public async Task<Package> GeneratePackageAsync(string id, PackageArchiveReader nugetPackage, PackageStreamMetadata packageStreamMetadata, User user, bool commitChanges)
         {
-            _reservedNamespaceService.IsPushAllowed(id, user, out IReadOnlyCollection<ReservedNamespace> userOwnedNamespaces);
+            var isPushAllowed = _reservedNamespaceService.IsPushAllowed(id, user, out IReadOnlyCollection<ReservedNamespace> userOwnedNamespaces);
 
-            var shouldMarkIdVerified = userOwnedNamespaces != null && userOwnedNamespaces.Any();
+            var shouldMarkIdVerified = isPushAllowed && userOwnedNamespaces != null && userOwnedNamespaces.Any();
```
Comment (skofman1) [inline src/NuGetGallery/Services/PackageUploadService.cs]: is there a UT covering this case?

File: src/NuGetGallery/Services/PackageUploadService.cs [hunk 2]
```diff
@@ -24,15 +24,15 @@ public class PackageUploadService: IPackageUploadService
...
             var package = await _packageService.CreatePackageAsync(
                 nugetPackage,
                 packageStreamMetadata,
                 user,
                 isVerified: shouldMarkIdVerified,
-                commitChanges: false);
+                commitChanges: commitChanges);
```
Comment (skofman1) [inline src/NuGetGallery/Services/PackageUploadService.cs]: and this case?

File: src/NuGetGallery/Services/PackageUploadService.cs (hunk 1 above)
Comment (shishirx34) [reply src/NuGetGallery/Services/PackageUploadService.cs]: added.

File: src/NuGetGallery/Services/PackageUploadService.cs (hunk 2 above)
Comment (shishirx34) [reply src/NuGetGallery/Services/PackageUploadService.cs]: added

**Pull request merged**